pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -656,9 +656,23 @@ func (service *SecurityPolicyService) deleteSecurityPolicy(obj interface{}, isVp
 	finalSecurityPolicyCopy.Rules = nsxSecurityPolicy.Rules
 
 	if isVpcEnabled(service) || isVpcCleanup {
-		vpcInfo, err := service.getVpcInfo(spNameSpace)
-		if err != nil {
-			return err
+		var vpcInfo *common.VPCResourceInfo
+		if isVpcCleanup == false {
+			vpcInfo, err = service.getVpcInfo(spNameSpace)
+			if err != nil {
+				return err
+			}
+		} else {
+			// In cleanup process, vpcInfo should be listed directly from security policy store to avoid calling VPC service.
+			// Get orgId, projectId, vpcId from security policy path "/orgs/<orgId>/projects/<projectId>/vpcs/<vpcId>/security-policies/<spId>"
+			if nsxSecurityPolicy.Path == nil {
```

Comment:
I'm not sure I understand this correct. If NSX security policy is created in NSX successfully by API call.  For cleanup, there must be a valid path by which we can delete it. 
For normal case, I think it means there is SP created in NSX side, and after operator restated from crash, it will re-initialized NSX security policy from nsx side and save it in the store. 
Meanwhile, opreator will rebuild a NSX security policy from CR and compare it with the existing one in the store. but the SP UID should be the same not changed since CR is not changed. At last, operator will re-sync with NSX to make both sides consistency.

Reply:
My bad, so the path is the NSX security policy object's path, not the path in security policy CR status. Ignore the previous comment. It's ok.